app-modules/ai/database/migrations/2025_05_19_214332_seed_permissions_add_research_advisor_permission.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2025, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use CanyonGBS\Common\Database\Migrations\Concerns\CanModifyPermissions;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    use CanModifyPermissions;
+
+    /** @var array<string, string> */
+    private array $permissions = [
+        'research_advisor.view-any' => 'Research Advisor',
+        'research_advisor.create' => 'Research Advisor',
+        'research_advisor.*.view' => 'Research Advisor',
+        'research_advisor.*.update' => 'Research Advisor',
+        'research_advisor.*.delete' => 'Research Advisor',
+        'research_advisor.*.restore' => 'Research Advisor',
+        'research_advisor.*.force-delete' => 'Research Advisor',
```

Comment:
The permission matrix is going to include checkboxes for each of these, but we don't use many of them. For example, you can still create research requests without the Create permission, you can rename and move them without Update, and you can delete them without Delete.

This is consistent behaviour with the institutional advisor, except I don't think that we seed all these extra permissions for that feature.

@Orrison should we opt to only seed `view-any` and `view` in this case?

Reply:
@danharrin yeah, I am fine with that. If we end up needing the permissions later we can add them then.

@krlmrr please remove any of the permissions here we are not currently using.